packages/matchbox/src/components/Popover/Popover.tsx

```diff
@@ -1,16 +1,68 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import { padding, layout } from 'styled-system';
-import { createPropTypes } from '@styled-system/prop-types';
+import { PaddingProps, LayoutProps, ResponsiveValue } from 'styled-system';
 import { Box } from '../Box';
 import PopoverOverlay from './PopoverOverlay';
 import PopoverContent from './PopoverContent';
 import { onKey, onKeys } from '../../helpers/keyEvents';
 import useWindowEvent from '../../hooks/useWindowEvent';
-import { deprecate } from '../../helpers/propTypes';
 import { findFocusableChild } from '../../helpers/focus';
-
-const Popover = React.forwardRef(function Popover(props, ref) {
+import type * as Polymorphic from '../../helpers/types';
+
+type BaseProps = PaddingProps &
+  LayoutProps & {
+    id?: string;
+    /**
+     * A React component to will trigger the popover
+     * Click event is handled for you if this component is uncontrolled.
+     */
+    trigger?: React.ReactNode;
+    /**
+     * Adds a padding to the Popover
+     * @deprecated Use padding system props instead
+     */
+    sectioned?: boolean;
+    /**
+     * Opens the popover.
+     * By default, open state is handled automatically. Passing this value in will turn this into a controlled component.
+     */
+    open?: boolean;
+    /**
+     * @deprecated Use `position` instead
+     */
+    left?: boolean;
+    /**
+     * @deprecated Use `position` instead
+     */
+    right?: boolean;
+    /**
+     * @deprecated Use `position` instead
+     */
+    top?: boolean;
+    /**
+     * @deprecated Use `position` instead
+     */
+    bottom?: boolean;
+    position?: ResponsiveValue<'topLeft' | 'topRight' | 'bottomLeft' | 'bottomRight'>;
+    /**
+     * Callback function that is called when clicking outside the popover, or hitting escape.
+     */
+    onClose?: (any) => void;
```

Comment:
Looks like this could either be a keydown or click event. I think `(e: Event) => void;` would work